pagefaultgames/pokerogue
# Taunt Implementation

## What are the changes?
This PR implements Taunt and Taunt handling logic around currently implemented use/edge cases.

## Why am I doing these changes?
I wanted a fun first PR to contribute to this awesome game and a friend asked me to implement Taunt.  

## What did change?
I have added implemented the Taunt move with logic alongside the conditions that that would cause it fail.  I've also added the (english) dialogue for all of the taunt messages and interactions.

### Screenshots/Videos
Will try to add these shortly. 

## How to test the changes?
* Run locally and start a new game with all Sinnoh starters
* Exclusively take lures to enhance chance of double battles
* Level Chimchar to 9 and learn Taunt
* Taunt will work on enemies but in double battle you can taunt your ally and test a lot better
* Try to use status moves and they should fail
* You can also test that Taunt leaves on swap

## Checklist
